### ISSUE TITLE
More elegant API for defining complex fuzzy rules

### ISSUE THREAD

**Issue by JDWarner** [scikit-fuzzy/scikit-fuzzy#75]
Some complex rules cannot be defined in the current framework. A minimal example is

IF `var0['good']` AND (`var1['bad']` OR `var2['good']`) THEN `consequent['happy']`

The issue is order of operations combined with user-defined groupings `()`.

One potential path forward would include overriding Python's `__and__`, and `__or__` object methods for fuzzy variables. These would return a new `CompositeVariable` object which would track order of operations of multiple constituent fuzzy variables. Done properly, this would allow an elegant API like

``` python
rule1 = fuzz.control.FuzzyRule()
rule1.antecedents = var0['good'] & (var1['bad'] | var2['good'])
rule1.consequents = consequent['happy']
```

The current method of defining rules _could_ be maintained, as it works well for simpler systems. However, the new API would be general enough to support nearly all use cases. We could simply move to the new API.

Comments welcome.

**Comment by jsexauer** [comment]
I like this idea a lot.  Looking ahead to eventual IEC 61131-7 (the fuzzy control language) implementation, here are a few things that I see that would need to be worked through:
 - What if the user wants to implement AND and OR using different algorithms (ex: OR is commonly implemented using MAX, ASUM, and BSUM).  How would we want to handle this?  My gut instinct is via 
an option in the Rule object
- We should also consider how to implement custom activation modes and accumulation modes in a rule.

See http://www.fuzzytech.com/binaries/ieccd1.pdf pages 13-14

**Comment by jsexauer** [comment]
Also, what do you think of using "~" for not.  So you could use
```
rule.antecedent = ~service['good'] | ~quality['good']
```

**Comment by JDWarner** [comment]
@jsexauer Love it! 

I feel like this is the right path, _extremely_ expressive and intuitive.

**Comment by JDWarner** [comment]
Closed by #79.
